packages/opentelemetry-instrumentation-http/src/types.ts

```diff
@@ -81,6 +89,8 @@ export interface HttpInstrumentationConfig extends InstrumentationConfig {
   requestHook?: HttpRequestCustomAttributeFunction;
   /** Function for adding custom attributes before response is handled */
   responseHook?: HttpResponseCustomAttributeFunction;
+  /** Function for adding custom attributes before a span is started */
+  startSpanHook?: StartSpanCustomAttributeFunction;
```

Comment:
I think it would be better to add two hooks here, one for incoming and one for outgoing requests.

Reply:
yup, thanks~two hooks here seems more friendly to user~